### ISSUE TITLE
[Support] TensorRT invalid key -17

### ISSUE THREAD

**Issue by tkffaul** [blakeblackshear/frigate#9608]
### Describe the problem you are having

I've read up on this... but i still dont understand fully. I have frigate running on a Xeon E5 v2 in LXC -> Docker. I have gone over the config 100 different ways... If I load the same config on version 12 i have no issues... but a minute after loading 13 i get:
frigate.record.maintainer      WARNING : Unable to keep up with recording segments in cache for <Pick whichever random camera>. Keeping the 6 most recent segments out of 7 and discarding the rest...

I get that it could be a speed issue (transfer)... but if it is, how does it work with the same docker and same LXC with version 12? The storage device is NFS and its local... i've tested the throughput and its big. Its out main storage server with a dedicated drive and pool just for this container. I have 8gb of Ram and 8 CPU cores with an nvidia card.... I just cant see how this is possible, so i'm checking to see if its a bug? I have 10 cameras most issues i see are over 30... 

### Version

0.13.1

### Frigate config file

```yaml
My config is huge and i dont feel like scrubbing it, if you want to see a section just ask and I'll post it.
```

### Relevant log output

```shell
_
```

### FFprobe output from your camera

```shell
_
```

### Operating system

Proxmox

### Install method

Docker CLI

### Coral version

Other

### Network connection

Wired

### Camera make and model

Reolink

**Comment by NickM-27** [comment]
as part of the rework on recordings we discovered a number of bugs in 0.12 that would lead to recordings silently being deleted when they shouldn't in 0.12. These have been fixed in 0.13 but that can lead to this issue occurring where it was masked before.

at a minimum would want to see a full screenshot of the system page and logs with `frigate.record.maintainer` set to debug

**Comment by tkffaul** [comment]
Not a Problem:
Its going to take a couple screen shots for the system page, anything you want in particular?

Here is a snip out of the log... the speeds are all over the place!

`2024-02-02 14:02:29.828001850  [2024-02-02 14:02:29] frigate.record.maintainer      DEBUG   : Copied /media/frigate/recordings/2024-02-02/19/Warehouse_Rear/02.14.mp4 in 0.4718739986419678 seconds.

2024-02-02 14:02:30.008517676  [2024-02-02 14:02:30] frigate.record.maintainer      DEBUG   : Copied /media/frigate/recordings/2024-02-02/19/Yard/02.14.mp4 in 0.4629700183868408 seconds.

2024-02-02 14:02:30.242060239  [2024-02-02 14:02:30] frigate.record.maintainer      DEBUG   : Copied /media/frigate/recordings/2024-02-02/19/Office_Front/02.13.mp4 in 0.47750306129455566 seconds.

2024-02-02 14:02:39.871840204  [2024-02-02 14:02:39] frigate.record.maintainer      DEBUG   : Copied /media/frigate/recordings/2024-02-02/19/Storage_Rear/02.25.mp4 in 1.381500005722046 seconds.

2024-02-02 14:02:39.873524127  [2024-02-02 14:02:39] frigate.record.maintainer      DEBUG   : Copied /media/frigate/recordings/2024-02-02/19/Warehouse/02.25.mp4 in 1.1140990257263184 seconds.

2024-02-02 14:02:39.874704077  [2024-02-02 14:02:39] frigate.record.maintainer      DEBUG   : Copied /media/frigate/recordings/2024-02-02/19/Office_Parking/02.24.mp4 in 0.9019098281860352 seconds.

2024-02-02 14:02:39.875980047  [2024-02-02 14:02:39] frigate.record.maintainer      DEBUG   : Copied /media/frigate/recordings/2024-02-02/19/Warehouse_Rear/02.24.mp4 in 0.6692240238189697 seconds.

2024-02-02 14:02:39.968057261  [2024-02-02 14:02:39] frigate.record.maintainer      DEBUG   : Copied /media/frigate/recordings/2024-02-02/19/Storage_Front/02.24.mp4 in 0.5750370025634766 seconds.

2024-02-02 14:02:40.042952338  [2024-02-02 14:02:40] frigate.record.maintainer      DEBUG   : Copied /media/frigate/recordings/2024-02-02/19/Yard/02.24.mp4 in 0.4578261375427246 seconds.

2024-02-02 14:02:40.285648107  [2024-02-02 14:02:40] frigate.record.maintainer      DEBUG   : Copied /media/frigate/recordings/2024-02-02/19/Office_Front/02.23.mp4 in 0.4895501136779785 seconds.

2024-02-02 14:02:40.401544070  [2024-02-02 14:02:40] frigate.record.maintainer      DEBUG   : Copied /media/frigate/recordings/2024-02-02/19/Yard2/02.23.mp4 in 0.5660190582275391 seconds.`

**Comment by NickM-27** [comment]
we will need a lot more logs than that. Preferably a full set of logs starting when frigate started and ending when the "unable to keep up" begins

the whole system page would be good to see

**Comment by NickM-27** [comment]
looks like tensorrt is not setup correctly

```
2024-02-02 13:19:33.320871804  Process camera_processor:Office_Front:

2024-02-02 13:19:33.323633275  Traceback (most recent call last):

2024-02-02 13:19:33.323675068    File "/usr/lib/python3.9/multiprocessing/process.py", line 315, in _bootstrap

2024-02-02 13:19:33.323679458      self.run()

2024-02-02 13:19:33.323697860    File "/usr/lib/python3.9/multiprocessing/process.py", line 108, in run

2024-02-02 13:19:33.323701737      self._target(*self._args, **self._kwargs)

2024-02-02 13:19:33.323705253    File "/opt/frigate/frigate/video.py", line 436, in track_camera

2024-02-02 13:19:33.323717252      process_frames(

2024-02-02 13:19:33.323721845    File "/opt/frigate/frigate/video.py", line 689, in process_frames

2024-02-02 13:19:33.323725938      detect(

2024-02-02 13:19:33.323745420    File "/opt/frigate/frigate/video.py", line 474, in detect

2024-02-02 13:19:33.323749320      region_detections = object_detector.detect(tensor_input)

2024-02-02 13:19:33.323753096    File "/opt/frigate/frigate/object_detection.py", line 225, in detect

2024-02-02 13:19:33.323756653      (self.labels[int(d[0])], float(d[1]), (d[2], d[3], d[4], d[5]))

2024-02-02 13:19:33.323766969  KeyError: -17
```

this usually happens when the nvidia driver is not updated to 535

**Comment by tkffaul** [comment]
![image](https://github.com/blakeblackshear/frigate/assets/55748571/b14d6627-5dcb-4392-9e23-82e594a2f609)
 it is 535

**Comment by NickM-27** [comment]
@tkffaul can you show the detector and model sections of your config

**Comment by tkffaul** [comment]
i use a quadro because of the encode/decode limitations on the consumer side.

**Comment by tkffaul** [comment]
it did do a fresh compile of the model at install, just an FYI

**Comment by tkffaul** [comment]
I see there is a newer driver, do you want me to upgrade? 535.154.05... I'm also running this same setup in another spot with a P4000 with the 535.129.03 driver without issue. So i assumed it wasn't a driver issue.

**Comment by NickM-27** [comment]
interesting, I am not sure why there would be a difference

**Comment by tkffaul** [comment]
Thats it! I updated the driver and it looks rock solid now! Hope this helps someone else! I'll just push them all up since this looks good!

**Comment by tkffaul** [comment]
I was wrong... dang... it looked so good. It just went a little longer.

**Comment by hargrove4570** [comment]
I've been tracking these same errors with the opt/frigate/frigate/object_detection.py", line 225, I wonder if it is very sensitive to the latest driver for older nvidia, most running compute 5.0. Adding another data point before opening my own ticket.

**Comment by hargrove4570** [comment]
Hey can you check you Nvidia container toolkit version with `nvidia-ctk --version`

Mine is 1.13.5 and I get a similar error wi th compute 5.0 and driver 535.54.03 on a GTX 750 ti. Given the refactor of tensorrt I wonder if this is actually the culprit that is seemingly fixed by going to the latest driver (unsure if that installs the latest runtime)

**Comment by nordeep** [comment]
Came here with the same issue: `KeyError: -17`
On the Frigate 0.12 I used tensor model `yolov7-tiny-416` on 0.13.1 Frigate can't start with this model.
`nvidia-smi` output:
```
Sun Feb  4 17:39:54 2024
+---------------------------------------------------------------------------------------+
| NVIDIA-SMI 545.23.08              Driver Version: 545.23.08    CUDA Version: 12.3     |
|-----------------------------------------+----------------------+----------------------+
| GPU  Name                 Persistence-M | Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |         Memory-Usage | GPU-Util  Compute M. |
|                                         |                      |               MIG M. |
|=========================================+======================+======================|
|   0  NVIDIA GeForce GTX 750 Ti      On  | 00000000:01:00.0 Off |                  N/A |
| 42%   47C    P0               2W /  38W |   1110MiB /  2048MiB |     15%      Default |
|                                         |                      |                  N/A |
+-----------------------------------------+----------------------+----------------------+

+---------------------------------------------------------------------------------------+
| Processes:                                                                            |
|  GPU   GI   CI        PID   Type   Process name                            GPU Memory |
|        ID   ID                                                             Usage      |
|=======================================================================================|
|    0   N/A  N/A    195175      C   frigate.detector.cuda                       104MiB |
|    0   N/A  N/A    195204      C   ffmpeg                                       71MiB |
|    0   N/A  N/A    195214      C   ffmpeg                                      104MiB |
|    0   N/A  N/A    195221      C   ffmpeg                                       39MiB |
|    0   N/A  N/A    197601      C   uwsgi                                       783MiB |
+---------------------------------------------------------------------------------------+
```

`nvidia-ctk --version` output:
```
NVIDIA Container Toolkit CLI version 1.14.4
commit: d167812ce3a55ec04ae2582eff1654ec812f42e1
```

For now managed to run without error model `yolov4-tiny-416` but it doesn't detect any persons.

**Comment by hargrove4570** [comment]
@nordeep yes my detection is completely whack even though before it was perfect. When I have time I'm going to find the old docker release that was actually good so that rolling back permanently is an option.

I'm guessing there was some major changes in the new tensorrt that doesn't support this card even though it meets the driver and compute version. When I run the container separately it gives an ERROR that GTX 750 Ti is detected but is not supported. Right now I'm waiting for my streams to stop like they did last night.

**Comment by nordeep** [comment]
@hargrove4570 for myself i decided to go back to 0.12.1 version which work perfectly or me.
```
services:
  frigate:
    container_name: frigate
    privileged: true # this may not be necessary for all setups
    restart: unless-stopped
    image: ghcr.io/blakeblackshear/frigate:0.12.1-tensorrt
```

**Comment by nordeep** [comment]
As I can see the changes haven't been significant from 0.12.1 at tensorrt docker container - https://github.com/blakeblackshear/frigate/tree/dev/docker/tensorrt

**Comment by hargrove4570** [comment]
Did you just delete frigate.db or have to do anything else to roll back? I'd did it late last night and noticed a memory error which is why I tried to get 0.13.1 going again.

**Comment by nordeep** [comment]
@hargrove4570 I just gave up on the record archive. It was fine to me. frigate.db was re-created.

**Comment by tkffaul** [comment]
I also rolled back to 12.1, but i ordered a p400 to test on this. I think you guys are on to something with the compute 5.0, I just don't know enough about cuda to be useful trouble shooting.

**Comment by tkffaul** [comment]
So same install, same driver, just pulled the K2200 and put in a P400. Works perfect, there must be something with the older 5.0 cards.

**Comment by NickM-27** [comment]
Thanks for updating, that is good info.

**Comment by qubex22** [comment]
> So same install, same driver, just pulled the K2200 and put in a P400. Works perfect, there must be something with the older 5.0 cards.

Had the same problem with a K620, just bought a P600 and works well too

**Comment by KastB** [comment]
Same issues for my GTX850M - is there any hope to get this fixed with frigate 0.13 or should I try downgrading as well?
Is there any info we could provide / anything to test to help sorting this out?
